src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx

```diff
@@ -25,6 +25,47 @@ interface IstioConfigDetailsState {
 class IstioConfigDetailsPage extends React.Component<RouteComponentProps<IstioConfigId>, IstioConfigDetailsState> {
   aceEditorRef: React.RefObject<AceEditor>;
 
+  static fetchIstioObjectDetailsFromProps = (props: IstioConfigId) => {
+    let promiseConfigDetails = API.getIstioConfigDetail(
+      authentication(),
+      props.namespace,
+      props.objectType,
+      props.object
+    );
+    let promiseConfigValidations = API.getIstioConfigValidations(
+      authentication(),
+      props.namespace,
+      props.objectType,
+      props.object
+    );
+    const allPromise = Promise.all([promiseConfigDetails, promiseConfigValidations])
+      .then(([resultConfigDetails, resultConfigValidations]) => {
+        const newState: IstioConfigDetailsState = {
+          istioObjectDetails: resultConfigDetails.data,
+          validations: resultConfigValidations.data
+        };
+        return newState;
+      })
+      .catch(error => {
+        MessageCenter.add(API.getErrorMsg('Could not fetch IstioConfig details.', error));
+      });
+    return allPromise;
+  };
+
+  static getDerivedStateFromProps(props: RouteComponentProps<IstioConfigId>, currentState: IstioConfigDetailsState) {
```

Comment:
Does that work? How is the caller of `getDerivedStateFromProps` able to get the asynchronous result?

Maybe here it's better to use `componentDidUpdate` instead of `getDerivedStateFromProps`, since we're fetching the API, no?

Reply:
I don't know I am still testing if this work. I read that should work with asynchronous results, if not we should move to componentDidUpdate. This is why I did 2 PR for this tasks

Reply:
Ok,
According to the doc it seems recommended to use componentDidUpdate in such case : https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops . It says getDerivedStateFromProps is sort of a last option if other options don't work

Reply:
Yes, thanks @jotak